src/Server/TCPHandler.cpp

```diff
@@ -1417,6 +1423,19 @@ void TCPHandler::receiveClusterNameAndSalt()
     readStringBinary(salt, *in, 32);
 }
 
+#if USE_SSL
+void TCPHandler::generateAndSendNonce()
+{
+    /// Contains lots of stuff (including time), so this should be enough for NONCE.
+    pcg64_fast rng(randomSeed());
+    UInt64 rand = rng();
```

Comment:
It's equivalent to `UInt64 rand = randomSeed()` (yes, `rng` will return a different number, but it's deterministic). So why do we need `rng` here? And why `_fast`? Also the most random component of the seed is `tv_nsec` which is not so random.

Reply:
Seems like `thread_local_rng` is more random because:
 - it's initialized with `randomSeed()` as well, but the number of `thread_local_rng()` calls is an additional source of randomness (and for `rng` it's always 1)
 - it's `pcg64` without `_fast` suffix

Reply:
Agree, rebased.